src/segmentation.py

```diff
@@ -373,13 +383,36 @@ def extract_electrophysiology_intensities(full_manifest: dict , session: dict):
     rprint("="*80 + "\n")
 
 def match_masks(stack1_masks_path: np.ndarray, stack2_masks_path: np.ndarray) -> dict:
-    '''
-    Input should be 2 stacks of masks where stack1 is registered to stack2.
-    '''
-
     stack1_masks = tif_imread(stack1_masks_path)
-    stack2_masks= tif_imread(stack2_masks_path)
+    stack2_masks = tif_imread(stack2_masks_path)
     
+    # Add dimension debugging
+    print(f"Stack1 (2P) shape: {stack1_masks.shape}")
+    print(f"Stack2 (HCR) shape: {stack2_masks.shape}")
+    
+    # Check if dimensions match
+    # Replace your current dimension checking code with this:
+    if stack1_masks.shape != stack2_masks.shape:
```

Comment:
@jonnasingh how is it possible that they are in different sizes? Does it not mean there there is a bug somewhere?